test/src/unit-cppapi-deletes.cc

```diff
@@ -1921,3 +1928,216 @@ TEST_CASE_METHOD(
   remove_sparse_array();
 }
 #endif
+
+TEST_CASE_METHOD(
+    DeletesFx,
+    "CPP API: Deletion of invalid group writes",
+    "[cppapi][deletes][group][invalid]") {
+  create_dir(GROUP_NAME);
+
+  // Create and open group in write mode
+  tiledb::Group::create(ctx_, GROUP_NAME);
+  tiledb::Group group(ctx_, std::string(GROUP_NAME), TILEDB_WRITE);
+
+  // Try to delete group
+  REQUIRE_THROWS_WITH(
+      group.delete_group(GROUP_NAME),
+      Catch::Matchers::ContainsSubstring(
+          "Query type must be MODIFY_EXCLUSIVE"));
+  group.close();
+
+  // Try to delete group after close
+  REQUIRE_THROWS_WITH(
+      group.delete_group(GROUP_NAME),
+      Catch::Matchers::ContainsSubstring("Group is not open"));
+
+  remove_dir(GROUP_NAME);
+}
+
+TEST_CASE_METHOD(
+    DeletesFx,
+    "CPP API: Deletion of group writes",
+    "[cppapi][deletes][group]") {
+  create_dir(GROUP_NAME);
+
+  // Create group
+  tiledb::Group::create(ctx_, GROUP_NAME);
+  std::string array_path = tiledb::sm::URI(GROUP_NAME + "array/").to_string();
+  create_simple_array(array_path);
+
+  // Set expected
+  std::vector<tiledb::Object> group_expected = {
+      tiledb::Object(tiledb::Object::Type::Array, array_path, std::nullopt),
+  };
+
+  // Write to group
+  tiledb::Group group(ctx_, GROUP_NAME, TILEDB_WRITE);
+  group.add_member(array_path, false);
+  int32_t v = 123;
+  group.put_metadata("test_deletes_meta", TILEDB_INT32, 1, &v);
+  group.close();
+  group.open(TILEDB_WRITE);
+  group.put_metadata("test_deletes_meta", TILEDB_INT32, 1, &v);
+  group.close();
+  group.open(TILEDB_WRITE);
+  group.put_metadata("test_deletes_meta", TILEDB_INT32, 1, &v);
+  group.close();
+
+  // Add extraneous file
+  std::string extraneous_file_path = GROUP_NAME + "extraneous_file";
+  vfs_.touch(extraneous_file_path);
+
+  // Validate group structure
+  group.open(TILEDB_READ);
+  std::vector<tiledb::Object> group_received = read_group(group);
+  REQUIRE_THAT(
+      group_received, Catch::Matchers::UnorderedEquals(group_expected));
+  group.close();
+
+  // Validate group data
+  REQUIRE(vfs_.is_file(extraneous_file_path));
+  REQUIRE(vfs_.is_file(GROUP_NAME + tiledb::sm::constants::group_filename));
+  auto group_detail_dir =
```

Comment:
See my comment below on how we could create a GroupDetailsDirectory class...

Reply:
This is still a `#TODO` - I've added the `FileCount` class but need to make adjustments for Group support

Reply:
Marking as resolved for won't-implement after offline discussion.